### ISSUE TITLE
[declarations] does not work

### ISSUE THREAD

**Issue by maximeg** [facebook/flow#7350]
TLDR: `[declarations]` section behavior does not work as expected from the description in the documentation.

**The issue**

After each upgrade of flow, I found myself having errors appearing for some of my dependencies. Taking the example of `react-select`, I have now 47 errors from it (with flow 0.90).

This is outside of the scope of my code, and will eventually get fixed, so I looked into how to put those into warnings or to suppress them. 

I found https://flow.org/en/docs/config/declarations/ :
> The [declarations] section in a .flowconfig file tells Flow to parse files matching the specified regular expressions in declaration mode. In declaration mode the code is not typechecked. However, the signatures of functions, classes, etc are extracted and used by the typechecker when checking other code.

Perfect, that exactly what I want: ignoring the errors and having the types used in my app.
And I don't want to ignore them (in the `[ignore]` section, i.e. having in the end all types transformed into `any`).

But, adding this:
```
[declarations]
<PROJECT_ROOT>/node_modules/.*
```

does not silence the errors.

And in fact increases the number of errors reported. I have now more than 75 errors for `react-select` (new errors appearing seems to be errors being suppressed in the source code of `react-select` with some `$FlowFixMe` comments).

Flow version: 0.90

(Do note that this is not an issue about the naming of "declarations". Please, refrain from polluting this report.)

**Comment by ZhangYiJiang** [comment]
This seems to mirror / duplicate https://github.com/facebook/flow/issues/6717. I see the same issue by the way - it's really weird how a feature for silencing errors is actually increasing them

**Comment by vicapow** [comment]
Closing as duplicate of #6717
